raft/server.cc

```diff
@@ -1678,12 +1688,13 @@ future<> server_impl::set_configuration(config_member_set c_new, seastar::abort_
 
     auto f = _non_joint_conf_commit_promise.emplace().promise.get_future();
     if (as) {
-        _non_joint_conf_commit_promise->abort = as->subscribe([this] () noexcept {
+        _non_joint_conf_commit_promise->abort = as->subscribe([this, cfg] () noexcept {
             // If we're inside this callback, the subscription wasn't destroyed yet.
             // The subscription is destroyed when the field is reset, so if we're here, the field must be engaged.
             SCYLLA_ASSERT(_non_joint_conf_commit_promise);
             // Whoever resolves the promise must reset the field. Thus, if we're here, the promise is not resolved.
-            std::exchange(_non_joint_conf_commit_promise, std::nullopt)->promise.set_exception(request_aborted{});
+            std::exchange(_non_joint_conf_commit_promise, std::nullopt)->promise.set_exception(
+                request_aborted(format("Aborted while setting configuration (at index: {}, config: {})", _fsm->log_last_conf_idx(), cfg)));
```

Comment:
This might happen long after the original `set_configuration` call. The configuration might have changed significantly from that moment. If you want to print the current configuration at the moment of the exception, you should get it again from `_fsm`. If you want to actually get the configuration that we were trying to set, it's more complex -- you'd have to obtain it from the added entry.

Also I'm not sure that we have a guarantee that `log_last_conf_idx()` is the same entry  that we added above. What if the exception is constructed after yet another configuration entry was added? Better to get the details from `e` (and you could print the term too)